### PR TITLE
fix: remove meta_type config

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix: remove meta_type config, Cannot set both embedded dir and [address|endpoints] config.

Closes #issue

### CODE REVIEW

File: docker/databend-query-docker.toml [hunk 1]
```diff
@@ -37,7 +37,6 @@ dir = "./.databend/logs"
 
 [meta]
 # To enable embedded meta-store, set address to ""
-embedded_dir = "_meta_embedded_1"
 address = "0.0.0.0:9191"
```
Comment (BohuTANG) [inline docker/databend-query-docker.toml:40]: Change all `address = "0.0.0.0:9191"` to `endpoints = ["0.0.0.0:9191"]`?

File: src/binaries/query/main.rs [hunk 2]
```diff
@@ -39,6 +39,7 @@ use tracing::info;
 #[databend_main]
 async fn main(_global_tracker: Arc<MemoryTracker>) -> common_exception::Result<()> {
     let conf: Config = Config::load()?;
+    println!("config: {:?}", conf);
```
Comment (BohuTANG) [inline src/binaries/query/main.rs:42]: Move this print info to the segment with `Print information to users.`?

File: src/query/config/src/outer_v0.rs [hunk 3]
```diff
@@ -1542,43 +1533,17 @@ pub struct MetaConfig {
...
-
-        // Fallback is used for forward compatbility, that is:
-        // First check embedded config, then endpoints, finally address.
-        if t == MetaType::Fallback {
-            if !self.embedded_dir.is_empty() && self.endpoints.is_empty() {
-                self.meta_type = MetaType::Embedded.to_string();
-            } else {
-                self.meta_type = MetaType::Remote.to_string();
-            }
+        println!(
```
Comment (BohuTANG) [inline src/query/config/src/outer_v0.rs:1536]: This `println` is for debug?

File: src/query/config/src/inner.rs [hunk 4]
```diff
@@ -264,13 +243,12 @@ pub struct MetaConfig {
     /// Certificate for client to identify meta rpc serve
     pub rpc_tls_meta_server_root_ca_cert: String,
     pub rpc_tls_meta_service_domain_name: String,
-    pub meta_type: MetaType,
 }
 
 impl Default for MetaConfig {
     fn default() -> Self {
         Self {
-            embedded_dir: "./.databend/meta_embedded".to_string(),
```
[the remark this answers is not in the document]
Comment (lichuang) [reply src/query/config/src/inner.rs:273]: `MetaType ` now has been removed, and cannot set bot embedded_dir and [address|endpoints ], if want to use embedded meta, user MUST set `embedded_dir` ifself.

**Comment by BohuTANG** [comment]
```
cargo-machete found the following unused dependencies in /workspace:
common-config -- /workspace/src/query/config/Cargo.toml:
	strum_macros
```

**Comment by BohuTANG** [comment]
Some test still failure, need to addressed

**Comment by drmingdrmer** [comment]
Shall we remove `address` too? It's still confusing to users with two config entries for service endpoints.

**Comment by Xuanwo** [comment]
> Shall we remove `address` too? It's still confusing to users with two config entries for service endpoints.

I'm second with this.

File: src/query/config/src/inner.rs [hunk 5]
```diff
@@ -264,29 +242,44 @@ pub struct MetaConfig {
...
+        let has_remote = !self.endpoints.is_empty();
+        if has_embedded_dir && has_remote {
+            return Err(ErrorCode::InvalidConfig(
+                "Cannot set both embedded dir and endpoints in meta config".to_string(),
+            ));
+        }
+
+        if !has_embedded_dir && !has_remote {
+            return Err(ErrorCode::InvalidConfig(
+                "Set embedded_dir or endpoints config in meta config".to_string(),
```
Comment (BohuTANG) [inline src/query/config/src/inner.rs:278]:
```suggestion
                "Please set your meta endpoints config: endpoints = ["<your-meta-service-endpoints>"]".to_string(),
```

File: src/query/config/src/inner.rs [hunk 6]
```diff
@@ -264,29 +242,44 @@ pub struct MetaConfig {
...
-        Ok(self.meta_type == MetaType::Embedded)
+        Ok(!self.embedded_dir.is_empty())
+    }
+
+    pub fn check_valid(&self) -> Result<()> {
+        let has_embedded_dir = !self.embedded_dir.is_empty();
+        let has_remote = !self.endpoints.is_empty();
+        if has_embedded_dir && has_remote {
+            return Err(ErrorCode::InvalidConfig(
+                "Cannot set both embedded dir and endpoints in meta config".to_string(),
```
Comment (BohuTANG) [inline src/query/config/src/inner.rs:272]:
```suggestion
                "Can not set embedded_dir and endpoints at the same time, embedded_dir is only for testing, please remove this config".to_string(),
```

File: src/binaries/query/main.rs [hunk 7]
```diff
@@ -44,6 +44,7 @@ async fn main(_global_tracker: Arc<MemoryTracker>) -> common_exception::Result<(
         return Ok(());
     }
 
+    conf.meta.check_valid()?;
```
Comment (BohuTANG) [inline src/binaries/query/main.rs:47]: Can we move this check to the config load?

Comment (BohuTANG) [reply src/binaries/query/main.rs:47]: Or add an new function `check_valid` for `Config`. Inside the `check_valid` we can do our all valid checking.

**Review by drmingdrmer — APPROVED** [review]

File: src/binaries/query/main.rs (hunk 7 above)
Comment (lichuang) [reply src/binaries/query/main.rs:47]: > Can we move this check to the config load?

If check when config loaded, some test may fail, cause `endpoints` and `embedded_dir` now has no default values.

Comment (BohuTANG) [reply src/binaries/query/main.rs:47]:
```
    let conf: Config = Config::load()?; -- Why test may fail if we check it in the end of load?

    if run_cmd(&conf) {
        return Ok(());
    }

    conf.meta.check_valid()?;
```

Comment (BohuTANG) [reply src/binaries/query/main.rs:47]: Like this:
https://github.com/BohuTANG/databend/commit/04cda338e866c42cc50b3365bc8102ddee0b5875

I think it's OK.

Comment (BohuTANG) [reply src/binaries/query/main.rs:47]: Or we can add a check_valid for Config.

File: src/query/config/src/inner.rs [hunk 8]
```diff
@@ -65,11 +65,19 @@ impl Config {
...
+        let cfg: Self = OuterV0Config::load()?.try_into()?;
 
+        // Only check meta config when cmd is empty.
+        if cfg.cmd.is_empty() {
+            cfg.meta.check_valid()?;
+        }
         Ok(cfg)
     }
 
+    pub fn load_with_configs(kvs: Vec<(String, Option<String>)>) -> Result<Self> {
```
Comment (BohuTANG) [inline src/query/config/src/inner.rs:77]: This is used for test ,can we move it out from inner.rs to test?

File: src/query/service/tests/it/sessions/session_context.rs [hunk 9]
```diff
@@ -23,7 +23,10 @@ use databend_query::sessions::SessionContext;
 
 #[tokio::test]
 async fn test_session_context() -> Result<()> {
-    let conf = Config::load()?;
+    let conf = Config::load_with_configs(vec![(
```
Comment (BohuTANG) [inline src/query/service/tests/it/sessions/session_context.rs:26]: If the load only affects this test, I think we can not call `conf.load` for it, build the Config as:
```
let conf = Config {....}
```

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
cc @ZhiHanZ @flaneur2020  for checking whether the cloud is using:
```
embedded_dir = "_meta_embedded_1"
or
address = "0.0.0.0:9191"
```
Otherwise, the query will stop when it starts.

**Pull request merged**